src/chk/chk_engine.c

```diff
@@ -669,28 +734,46 @@ chk_engine_start(uint64_t gen, uint32_t rank_nr, d_rank_t *ranks,
 	D_ASSERT(rank_list != NULL);
 	D_ASSERT(d_list_empty(&ins->ci_pool_list));
 	D_ASSERT(d_list_empty(&ins->ci_pending_list));
-	D_ASSERT(ins->ci_sched == ABT_THREAD_NULL);
 
-	if (ins->ci_iv_ns != NULL) {
-		ds_iv_ns_put(ins->ci_iv_ns);
-		ins->ci_iv_ns = NULL;
-	}
+	if (ins->ci_sched != ABT_THREAD_NULL)
+		ABT_thread_free(&ins->ci_sched);
 
-	if (ins->ci_iv_group != NULL) {
-		crt_group_secondary_destroy(ins->ci_iv_group);
-		ins->ci_iv_group = NULL;
+	chk_iv_ns_cleanup(&ins->ci_iv_ns);
+
+	if (chk_is_on_leader(gen, leader, true)) {
+		ins->ci_iv_ns = chk_leader_get_iv_ns();
+		if (unlikely(ins->ci_iv_ns == NULL))
+			goto out_log;
+	} else {
+		if (ins->ci_iv_group != NULL) {
+			crt_group_secondary_destroy(ins->ci_iv_group);
+			ins->ci_iv_group = NULL;
+		}
+
+		rc = crt_group_secondary_create(CHK_DUMMY_POOL, NULL, rank_list, &ins->ci_iv_group);
+		if (rc != 0)
+			goto out_log;
+
+		uuid_parse(CHK_DUMMY_POOL, dummy_pool);
```

Comment:
group name is string, then convert it to uuid?  any reason to do that?  is it unique?
BTW: I saw it is 000000 in test testing log?  is it expected?


might just create a global UUID, and use D_INFO to output it, to make sure this IV is unique?

Reply:
The second parameter of "ds_iv_ns_create" is "uuid_t" instead of "string", that is why we do uuid_parse. As for "000000" uuid in the log, we need to call "DF_UUIDF" that will dump the whole uuid_t instead of just the first 8 bytes.